coco-text/prepare_coco.py

```diff
@@ -0,0 +1,45 @@
+import coco_text
+import numpy as np
+import click
+
+
+@click.command()
+@click.option("--image_path", "-i", default="./images/", help="Directoy where the COCO-2014 images are located.", type=click.Path(exists=True))
+@click.option("--annotations", "-a", default="cocotext.v2.json", help="The json file that contains the annotations.", type=click.Path(exists=True))
+@click.option("--relative/--absolute", default=True, help="Whether the created imagefile should relative or absolute paths.")
+def prepare_coco(image_path, annotations, relative):
+    ct = coco_text.COCO_Text(annotations)
+
+    imgIds = ct.getImgIds(imgIds=ct.train, catIds=[('legibility', 'legible')])
+
+    imgs = ct.loadImgs(imgIds)
+
+    img_ids = [imgs[i]['id'] for i in range(len(imgs))]
+
+    if relative:
+        img_paths = ["./images/" + imgs[i]['file_name'] for i in range(len(imgs))]
+    else:
+        img_paths = [image_path + "/" + imgs[i]['file_name'] for i in range(len(imgs))]
+
+    np.savetxt("image_locations.txt", img_paths, fmt="%s")
+
+    bounding_boxes = []
+
+    for img_id in img_ids:
+        current_boxes = []
+        ann_ids = ct.getAnnIds(imgIds=img_id)
+        annos = ct.loadAnns(ann_ids)
+
+        for ann in annos:
+            box_w_h = ann['bbox']
```

Comment:
For increased readability and maintainability, I'd prefer the following syntax here:

```suggestion
            [bbox_x, bbox_y, bbox_w, bbox_h] = ann['bbox']
```

This would allow the following two lines to change to a more readable version (imo) as well:

```python
            box_tl_br = [(round(bbox_x), round(bbox_y)),
	                 (round(bbox_x + bbox_w), round(bbox_y + bbox_h))]
```